resources/images/model/images.go

```diff
@@ -45,42 +50,102 @@ func NewImagesModel(
 	}
 }
 
+// CreateImage parses artifact and uploads artifact file to the file storage - in parallel,
+// and creates image structure in the system.
+// Returns image ID and nil on success.
 func (i *ImagesModel) CreateImage(
-	artifact io.Reader,
 	metaConstructor *images.SoftwareImageMetaConstructor,
-	metaArtifactConstructor *images.SoftwareImageMetaArtifactConstructor) (string, error) {
-
-	if metaConstructor == nil || metaArtifactConstructor == nil {
+	imageReader io.Reader) (string, error) {
+	if metaConstructor == nil {
 		return "", controller.ErrModelMissingInputMetadata
 	}
+	if imageReader == nil {
+		return "", controller.ErrModelMissingInputArtifact
+	}
+	imageID, artifactFileID, err := i.handleArtifact(metaConstructor, imageReader)
+	// try to remove artifact file from file storage on error
+	if err != nil {
+		i.fileStorage.Delete(artifactFileID)
```

Comment:
In case of failed cleanup it would be nice to be able to catch this in logs, maybe wrapping in error from Delete into `err` would be an option.